src/javascript/reducers/agreementList.js

```diff
@@ -23,135 +25,223 @@ type Action =
+      price: 1234,
+      buyer: "123.12.23.21",
+      seller: "1231.123.12.1231",
+      buyerPaid: true,
+      buyerPaidDate: 1493690737488,
+      sellerPaid: true,
+      sellerPaidDate: 1493690737497,
+      agreementComplete: false,
+      agreementCompleteDate: 0,
+      error: "",
+      canceled: false,
+      canceledDate: 1493690737506,
+      requestPending: false,
+    },
+      "123123412312412413": {
+      creationDate: 1493690737488,
+      position: "buyer",
+      price: 1235,
+      buyer: "123.12.23.21",
+      seller: "1231.123.12.1231",
+      buyerPaid: true,
+      buyerPaidDate: 1493690737489,
+      sellerPaid: true,
+      sellerPaidDate: 1493690737496,
+      agreementComplete: false,
+      agreementCompleteDate: 0,
+      error: "",
+      canceled: false,
+      canceledDate: 1493690737503,
+      requestPending: false,
+    },
   },
-  allAgreementIds: ["123123412312412412"],
+  allAgreementIds: ["123123412312412412","123123412312412413"],
+  sortKey: "creationDate",
+  sortKind: "descending",
+};
+
+// The logic for changing state either belongs here or in a thunk or sorted
+// repeatedly in the container
+// Descending: Date(New to Old), Price(Largest to smallest)
+function sortAllAgreementIds(state: AgreementListState, sortKey: string, sortKind: "descending" | "ascending"): Array<string> {
+
+  const allAgreementIds: Array<string> = state.allAgreementIds,
+        agreementsById: AgreementListType = state.agreementsById;
+
+  const sortedAgreements: Array<string> = allAgreementIds.sort((a, b) => {
+
+    const aAgreement: string = agreementsById[a][sortKey];
+    const bAgreement: string = agreementsById[b][sortKey];
+
+    if (sortKind === "ascending") {
+      if (aAgreement < bAgreement) return -1;
+      else if (aAgreement > bAgreement) return 1;
+      return 0;
+
+    } else if (sortKind === "descending") {
+      if (aAgreement > bAgreement) return -1;
+      else if (aAgreement < bAgreement) return 1;
+      return 0;
+    };
+    return 0;
+  });
+
+  return sortedAgreements;
+
 };
 
 export default function agreementListReducer(state: AgreementListState = INITIAL_STATE, action: Action): AgreementListState {
   // set agreementId, and check if payload passes an object with an error
-  // or passes just the agreement int. This could be a very long
-
+  // or passes just the agreement id
+  const agreementsById: AgreementListType = state.agreementsById;
   let agreementId: string = '';
+  let agreement: AgreementType;
 
-  // flow really doesn't like this ternary operator, makes me super sad
-  // if (action.payload) {
-  //   agreementId =  typeof action.payload === 'object' ?  action.payload.agreementId.toString(): action.payload.toString();
-  // }
-
-  if (typeof action.payload === 'object') {
+  // REALLY ugly way of making sure that agreementId is in the object, is there
+  // a better solution to this?
+  if (typeof action.payload === 'object' && action.payload.agreementId) {
+    //$FlowFixMe
```

Comment:
`$FlowFixMe` is the default `suppress_comment`, and is not overridden unless we specify any other `supress_comment`s.